fvm/src/kernel/default.rs

```diff
@@ -985,18 +985,20 @@ where
             .call_manager
             .charge_gas(self.call_manager.price_list().on_actor_event_validate(len))?;
 
-        let actor_evt = match panic::catch_unwind(|| {
-            fvm_ipld_encoding::from_slice(raw_evt).or_error(ErrorNumber::IllegalArgument)
-        }) {
-            Ok(v) => v,
-            Err(e) => {
-                log::error!("panic when decoding cbor from actor: {:?}", e);
-                Err(syscall_error!(IllegalArgument; "panic when decoding cbor from actor").into())
-            }
+        let actor_evt = {
+            let res = match panic::catch_unwind(|| {
+                fvm_ipld_encoding::from_slice(raw_evt).or_error(ErrorNumber::IllegalArgument)
+            }) {
+                Ok(v) => v,
+                Err(e) => {
+                    log::error!("panic when decoding event cbor from actor: {:?}", e);
+                    Err(syscall_error!(IllegalArgument; "panic when decoding event cbor from actor").into())
+                }
+            };
+            t.stop();
```

Comment:
Now capturing a reading even on deserialization errors.

Reply:
I'd be careful about that. We usually don't time _failures_ because it can mess up the number's we're looking for (e.g., we can stop half-way through parsing something).

Reply:
I think this makes sense here as long as the test harness is controlling for when it's sending malformed input and when it isn't. This allows us to calibrate the costs for invalid CBOR input, although that will be less relevant if/when we move away from the CBOR-based ABI.

Reply:
Yeah, fair enough. And the gas charges here are specifically about parsing.